examples/dataflow/chatbot/operations.py

```diff
@@ -0,0 +1,223 @@
+    message = message.replace("\n", "\\n")
+    # Replacing all double quotes with single quotes
+    # Because we are quoting 'message' in double quotes
+    message = message.replace('"', "'")
+    data = '{"text":' + f'"{message}"' + "}"
+
+    url = f"https://api.gitter.im/v1/rooms/{room_id}/chatMessages"
+    async with self.parent.session.post(
+        url, data=data, headers=headers
+    ) as resp:
+        response = await resp.json()
+        return
+
+
+@op(
+    inputs={"message": MESSAGE,},
+    outputs={"message": TOSEND},
+    config_cls=GitterChannelConfig,
+    imp_enter={"config": (lambda self: self.config.config)},
+    ctx_enter={"cfg": (lambda self: self.parent.config())},
+)
+async def interpret_message(self, message):
+    greet = ["hey", "hello", "hi"]
+    for x in greet:
+        if x in message.lower():
+            return {"message": "Hey Hooman ฅ^•ﻌ•^ฅ"}
+
+    def extract_data(raw_data):
+        raw_data = raw_data.split("data:")
+        data = {"model-data": raw_data[1]}
+        raw_data = raw_data[0].split("\n")
+        for x in raw_data:
+            k, *v = x.split(":")
+            if isinstance(v, list):  # for features
+                v = ":".join(v)
+            k = k.strip()
+            v = v.strip()
+            if k:  # avoid blank
+                data[k] = v
+        return data
+
+    # Removing username from message
+    # The regex matches @ followed by anything that
+    # is not a whitespace in the first group and
+    # the rest of the string in the second group.
+    # We replace the string by the second group.
+    message = re.sub(r"(@[^\s]+)(.*)", r"\2", message).strip()
+    access_token = await self.cfg.get("access_token")
+
+    if message.lower().startswith("train model"):
+        return {"message": "Gimme more details!!"}
+
+    elif message.lower().startswith("predict:"):
+        # only replace first occurence of predict
+        # because the feature to predict will be labeled predict
+        raw_data = message.replace("predict:", "", 1).strip()
+        cmds = ["predict", "all"]
+
+    elif message.lower().startswith("details:"):
+        raw_data = message.replace("details:", "",).strip()
+        cmds = ["train"]
+
+    else:
+        return {"message": " Oops ,I didnt get that ᕙ(⇀‸↼‶)ᕗ "}
+
+    # If predict or train, extract data
+    data = extract_data(raw_data)
+    if "model-type" in data:
+        model_type = data["model-type"]
+    if "model-name" in data:
+        model_name = data["model-name"]
+    else:
+        model_name = "mymodel"
+
+    features = data["features"].split(" ")
+    predict = data["predict"]
+    model_data = data["model-data"]
+    # Temperrorly write data to a csv file
+    temp_file = tempfile.NamedTemporaryFile(suffix=".csv", delete=False)
+    temp_file.write(model_data.lstrip().encode())
+    temp_file.seek(0)
+
+    stdout = io.StringIO()
+    with contextlib.redirect_stdout(stdout):
+        await CLI._main(
```

Comment:
We can skip the json decoding by just grabbing the records as the output of `.cli`

```suggestion
        preds = await CLI.cli(
```